src/styles/main.css

```diff
@@ -0,0 +1,423 @@
+  content: '23';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day--24::before {
+  content: '24';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day--25::before {
+  content: '25';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day--26::before {
+  content: '26';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day--27::before {
+  content: '27';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day--28::before {
+  content: '28';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day--29::before {
+  content: '29';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day--30::before {
+  content: '30';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day--31::before {
+  content: '31';
+  display: flex;
+  margin: auto;
+  align-items: center;
+  justify-content: center;
+  color: black;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day:hover {
+  cursor: pointer;
+  background-color: #ffbfcb;
+  transform: translateY(-20px);
+  transition: all 500ms linear;
+}
+
+.calendar--month-length-28 .calendar__day:nth-child(n + 29) {
+  display: none;
+}
+
+.calendar--month-length-29 .calendar__day:nth-child(n + 30) {
+  display: none;
+}
+
+.calendar--month-length-30 .calendar__day:nth-child(n + 31) {
+  display: none;
+}
+
+.calendar--month-length-31 .calendar__day:nth-child(n + 32) {
+  display: none;
+}
+
+.calendar--start-day-mon .calendar__day:first-child {
+  margin-left: calc(0px);
+}
+
+.calendar--start-day-tue .calendar__day:first-child {
+  margin-left: calc(101px);
+}
+
+.calendar--start-day-wed .calendar__day:first-child {
+  margin-left: calc(202px);
+}
+
+.calendar--start-day-thu .calendar__day:first-child {
+  margin-left: calc(303px);
+}
+
+.calendar--start-day-fri .calendar__day:first-child {
+  margin-left: calc(404px);
+}
+
+.calendar--start-day-sat .calendar__day:first-child {
+  margin-left: calc(505px);
+}
+
+.calendar--start-day-sun .calendar__day:first-child {
+  margin-left: calc(606px);
```

Comment:
The styles for `.calendar__day--8::before` to `.calendar__day--31::before` are repetitive. Consider using a single class for these styles and dynamically setting the content property using a data attribute or a similar approach to avoid repetition.